gazelle/internal/spreso/pin_store_v2.go

```diff
@@ -100,9 +100,10 @@ func NewPkgRefFromV2Pin(v2p *V2Pin) (*PackageReference, error) {
 }
 
 func NewPinStateFromV2PinState(ps *V2PinState) (PinState, error) {
-	if ps.Revision == "" {
-		return nil, fmt.Errorf("revision cannot be empty %+v", ps)
-	}
+	// revision is empty for registry pins
+	// if ps.Revision == "" {
+	// 	return nil, fmt.Errorf("revision cannot be empty %+v", ps)
+	// }
 	if ps.Version != "" {
 		return NewVersionPinState(ps.Version, ps.Revision), nil
 	}
```

Comment:
Could we check `ps.Version` first to circumvent this

Reply:
Yes! I reordered these and was able to keep the `ps.Revision == ""` guard.